src/permutation.jl

```diff
@@ -0,0 +1,65 @@
+export PermutationTest
+
+function ptstats(x,y,f)
+    xy = vcat(x,y)
+    rx = 1 : length(x)
+    ry = length(xy)-length(y)+1 : length(xy)
+    T = typeof(f(xy))  # is there a better way?  e.g., does a funtype() exist, like eltype()??
```

Comment:
Yes, a better way would be to use an array comprehension in the constructor below, so that the element type is computed automatically.

Reply:
i'm not sure what you mean.  can you be more specific?

Reply:
I mean drop the `for (i,p) in enumerate(P)` loop, and instead do `xsamples = [f(view(p, rx)) for p in P]`.

Reply:
ahah.  but...  is omitting a call to `f` really preferable to looping over `P` twice?  i'm also not sure how to similarly change the `shuffle` call in `ApproximatePermutationTest`.

Reply:
The advantage is not omitting a call to `f`, it's relying on the compiler to do inference for us. For example, if `f` returned elements of heterogeneous types (which admittedly isn't typical, but still), using the type of the first element wouldn't be correct. I'm not sure how costly going over `P` is, though.

For `ApproximatePermutationTest`, you could use a single comprehension and store an array of two-element tuples.